pkg/controller/identity/iamrolepolicyattachment/controller.go

```diff
@@ -125,90 +123,61 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 		}, nil
 	}
 
+	current := cr.Spec.ForProvider.DeepCopy()
+	iam.LateInitializePolicy(&cr.Spec.ForProvider, attachedPolicyObject)
+	if !reflect.DeepEqual(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
 	cr.SetConditions(runtimev1alpha1.Available())
 
-	cr.UpdateExternalStatus(*attachedPolicyObject)
+	cr.Status.AtProvider = iam.GenerateRolePolicyObservation(*attachedPolicyObject)
 
 	return managed.ExternalObservation{
-		ResourceExists: true,
+		ResourceExists:   true,
+		ResourceUpToDate: cr.Spec.ForProvider.PolicyARN == cr.Status.AtProvider.AttachedPolicyARN,
 	}, nil
 }
 
 func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
-	cr, ok := mgd.(*v1alpha3.IAMRolePolicyAttachment)
+	cr, ok := mgd.(*v1beta1.IAMRolePolicyAttachment)
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
 	}
 
 	cr.SetConditions(runtimev1alpha1.Creating())
 
-	req := e.client.AttachRolePolicyRequest(&awsiam.AttachRolePolicyInput{
-		PolicyArn: aws.String(cr.Spec.PolicyARN),
-		RoleName:  aws.String(cr.Spec.RoleName),
-	})
+	req := e.client.AttachRolePolicyRequest(iam.GenerateAttachRolePolicyInput(&cr.Spec.ForProvider))
+	req.SetContext(ctx)
```

Comment:
As I mentioned in IAMRole controller, you can call `Send` directly on the request and you don't need to set context as `Send` does that when you pass the context in.